tests/integration/ha/test_ha.py

```diff
@@ -78,6 +78,68 @@ async def test_kill_db_process(ops_test: OpsTest, request):
     logger.info("Killing leader process...")
     await helpers.send_control_signal(ops_test=ops_test, unit_name=leader_name, signal="SIGKILL")
 
+    # Check that process is down
+    assert await helpers.is_down(ops_test=ops_test, unit=leader_name)
+
+    await asyncio.sleep(RESTART_DELAY * 2)
```

Comment:
**question** I'm not sure I fully follow: shouldn't we check that the writing still continues when the process is down? I would remove this sleep and make sure that the writes have increased since line 76

Reply:
Yes, the delay was a small addition the kill_db_process was missing. But I realize this sleep is better suited to go after the increment check. Thanks!

Reply:
@deusebio Let me backtrack on what I said, I don't think we need to check that writes happen while unit is down, because the client needs to handle the `NoLeaderForPartition` that raises when leader goes down. There is a short time where writes will fail. _If_ the client is well constructed, those writes will be cached into a buffer to be sent when a new leader is elected. But to us there is no difference, as what's important is that service is still functional after the reelection, and previous leader rejoins nicely.